packages/node/src/transport/HttpInboundTransport.ts

```diff
@@ -51,11 +59,32 @@ export class HttpInboundTransport implements InboundTransport {
 
       try {
         const message = req.body
-        const encryptedMessage = JSON.parse(message)
-        await messageReceiver.receiveMessage(encryptedMessage, {
-          session,
+        const encryptedMessage = JSON.parse(message) as EncryptedMessage
+
+        const observable = agent.events.observable<AgentMessageProcessedEvent>(AgentEventTypes.AgentMessageProcessed)
+        const subject = new ReplaySubject(1)
+
+        observable
+          .pipe(
+            first((e) => e.type === AgentEventTypes.AgentMessageProcessed),
+            timeout({
+              first: 10000, // timeout after 10 seconds
+              meta: 'HttpInboundTransport.start',
+            })
+          )
+          .subscribe(subject)
```

Comment:
This won't work, as there could be multiple messages processed. I think we need to assign each message an unique id, or we can lookg at the encrypted message and include that in the event as well. But we need some identifier

Reply:
- Updated the `AgentMessageProcessedEvent` to have optional parameter of the `encryptedMessage`